src/neuroconv/datainterfaces/ecephys/baserecordingextractorinterface.py

```diff
@@ -72,6 +72,19 @@ def get_metadata(self):
 
         return metadata
 
+    def synchronize_starting_time(self, starting_time: float):
+        self.recording_extractor.set_times(times=self.recording_extractor.get_times() + starting_time)
+
+    def synchronize_timestamps(self, timestamps: ArrayType):
+        self.recording_extractor.set_times(times=timestamps)
+
+    def synchronize_with_pulses(self, pulse_timestamps: ArrayType):
+        self.recording_extractor.set_times(
+            times=infer_timestamps_from_pulses(
+                relative_timestamps=self.recording_extractor.get_times(), pulse_timestamps=pulse_timestamps
+            )
+        )
```

Comment:
```suggestion
    def synchronize_with_pulses(
        self,
        pulse_timestamps_parent: ArrayType,
        pulse_timestamps_child: ArrayType,
        timestamps_child: Optional[ArrayType] = None,
    ):
        if timestamps_child is None:
            timestamps_child = self.recording_extractor.get_times()
        self.recording_extractor.set_times(
            times=infer_timestamps_from_pulses(
                 pulse_timestamps_parent=pulse_timestamps_parent,
                 pulse_timestamps_child=pulse_timestamps_child
                timestamps_child=timestamps_child,
            )
        )
```

Reply:
Yeh, this one is still WIP

Reply:
I wouldn't say parent/child is the best way to refer to the relationship. They're systems acquiring in parallel, but we, or the experimenter, designates one to be the 'primary' timing system which then tracks reference times across the secondary systems.

I've adjusted the names here to reflect this primary/secondary notion, and also how if the timings from the secondary timing streams aren't reflective of exact event occurrences, they become 'reference times'.